xcode/Subconscious/Shared/Components/FirstRun/RecoveryPhraseView.swift

```diff
@@ -42,10 +63,61 @@ struct RecoveryPhraseView: View {
     }
 }
 
+enum RecoveryPhraseAction: Hashable {
+    case setPhrase(_ phrase: String)
+    case copy
+}
+
+struct RecoveryPhraseModel: ModelProtocol {
+    var phrase: String = ""
+    var didCopy: Bool = false
+
+    static let logger = Logger(
+        subsystem: Config.default.rdns,
+        category: "RecoveryPhrase"
+    )
+
+    static func update(
+        state: RecoveryPhraseModel,
+        action: RecoveryPhraseAction,
+        environment: Void
+    ) -> Update<RecoveryPhraseModel> {
+        switch action {
+        case .setPhrase(let phrase):
+            var model = state
+            model.phrase = phrase
+            return Update(state: model)
+        case .copy:
+            // Copy to clipboard
+            UIPasteboard.general.string = state.phrase
```

Comment:
Is `UIPasteboard.general` always available? I'm imagining running this function in isolation as part of a test

Reply:
This was a good shout-out. Tests did work previously, but would copy things to your clipboard while running.

I have updated the PR to introduce a RecoveryPhraseEnvironment which includes a mockable pasteboard that defaults to UIPasteboard.general. I have also added unit tests.